### ISSUE TITLE
Print fully qualified type names in type checker errors

### ISSUE THREAD

**Issue by missingdays** [astral-sh/ty#950]
Consider this example

```python
class A:
    class B:
        pass

class C:
    class B:
        pass


a: A.B = C.B()
```

Here, `ty` reports `Object of type B is not assignable to B`. This message does not help fixing the error.

This this example, it's easy to see the problem. However, consider if two `B` classes were coming from different packages. In that case, it would be much harder to understand why the error is happening.

Printing the fully qualified name, with packages, module and inner classes would improve the error message while keeping the error message clear.

**Comment by missingdays** [comment]
I will be happy to send a merge request for this. However, I couldn't find a proper way to get the fully qualified name of a type, with packages and all.

**Comment by MichaReiser** [comment]
Makes sense.

We should probably avoid doing this all the time and only print the full qualified name if the name is ambiguous in the current file (e.g. there are multiple symbols with the same name or the type isn't even imported)

**Comment by AlexWaygood** [comment]
Thanks for the report! I think we can fold this into #848, but this is a great additional example of why we need to improve our diagnostics
